Date Display Formatter/js/dateHandler.js

```diff
@@ -0,0 +1,231 @@
+                monthStr += date[startMonth[i]];
+            }
+        }
+        let month = parseInt(monthStr);
+
+        return month;
+
+    }
+
+    getYears(date, inputRegexObj){
+        let startYear = inputRegexObj.yearArr;
+
+        let yearStr = "";
+        for(let i = 0; i < 4; i++){
+            if(startYear[i] != -1){
+                yearStr += date[startYear[i]];
+            }else{
+                if(i == 0){
+                    yearStr += "2";
+                }else{
+                    yearStr += "0";
+                }
+                
+            }
+        }
+        let year = parseInt(yearStr);
+
+        return year;
+
+    }
+
+    getDateByRegex(date, inputRegexObj, outputReg){
+        let startDay = inputRegexObj.dayArr;
+        let startMonth = inputRegexObj.monthArr;
+        let startYear = inputRegexObj.yearArr;
+
+        let outputRegex = outputReg.split('');
+        let outputRegexObj = this.getDateIndexes(outputRegex);
+        let endDay = outputRegexObj.dayArr;
+        let endMonth = outputRegexObj.monthArr;
+        let endYear = outputRegexObj.yearArr;
+
+        for(let i = 3; i >= 0; i--){
+            if(endYear[i] != -1){
+                if(startYear[i] !=- 1){
+                    outputRegex[endYear[i]] = date[startYear[i]];
+                }else{
+                    if(i == 0){
+                        outputRegex[endYear[i]] = 2;
+                    }else{
+                        outputRegex[endYear[i]] = 0;
+                    }
+                    
+                }
+                
+            }
+        }
+        for(let i = 1; i >= 0; i--){
+            if(endMonth[i] != -1){
+                if(startMonth[i] !=- 1){
+                    outputRegex[endMonth[i]] = date[startMonth[i]];
+                }else{
+                    outputRegex[endMonth[i]] = 0;
+                }
+                
+            }
+        }
+        for(let i = 1; i >= 0; i--){
+            if(endDay[i] != -1){
+                if(startDay[i] !=- 1){
+                    outputRegex[endDay[i]] = date[startDay[i]];
+                }else{
+                    outputRegex[endDay[i]] = 0;
+                }
+                
+            }
+        }
+
+        return outputRegex.join('');
+    }
+
+    format(date, inputRegex, outputReg){
+
+        
+
+        if(inputRegex == "ms"){
+            let inputDate = new Date(parseInt(date));
+
+            switch (outputReg){
+                case "":
+                    let days = inputDate.getDate();
+                    let years = inputDate.getFullYear();
+                    let months = this.getMonthString(inputDate.getMonth()+1);
+                    
+                    return days + " " + months + " " + years;
+
+                case "from now":
+                    let nowDate = new Date();
+                    return Math.ceil(Math.abs(nowDate.getTime() - inputDate.getTime()) / (1000 * 3600 * 24)) +" days ago";
```

Comment:
It would be easier to read and understand that code if the calculation is moved to a separate function and string interpolation is used